src/huggingface_hub/fastai_utils.py

```diff
@@ -0,0 +1,403 @@
+    check_fastai_fastcore_versions()
+
+    os.makedirs(save_directory, exist_ok=True)
+
+    # saving config
+    # if user provides config then we update it with the fastai and fastcore versions in CONFIG_TEMPLATE.
+    if config is not None:
+        if not isinstance(config, dict):
+            raise RuntimeError(
+                f"Provided config should be a dict. Got: '{type(config)}'"
+            )
+        path = os.path.join(save_directory, CONFIG_NAME)
+        with open(path, "w") as f:
+            json.dump(config, f)
+
+    # creating README.md if none exist
+    _create_model_card(Path(save_directory))
+
+    # creating pyproject.toml if none exist
+    _create_model_pyproject(Path(save_directory))
+
+    # saving learner
+    # learner.export saves the model in `self.path/save_directory` and this folder should exist.
+    # We create this folder and call it `fastai_path`
+    fastai_path = os.path.join(learner.path, save_directory)
+    os.makedirs(fastai_path, exist_ok=True)
+    learner.export(
+        fname=os.path.join(save_directory, "model.pkl"),
+        pickle_protocol=DEFAULT_PROTOCOL,
+    )
+
+    # We move the model from `self.path/save_directory/model.pkl` to `save_directory/model.pkl`.
+    os.rename(
+        os.path.join(fastai_path, "model.pkl"),
+        os.path.join(save_directory, "model.pkl"),
+    )
+
+
+def from_pretrained_fastai(
+    model_id: str,
+    revision: Optional[str] = None,
+):
+    """
+    Load `model_id` files from the Hub.
+
+    Parameters:
+        model_id (:obj:`str`):
+            The model id where the pickled fastai.Learner is. Example: 'espejelomar/fastai-pet-breeds-classification'.
+        revision (:obj:`str`, `optional`):
+            Revision at which the repo's files are downloaded. See documentation of `snapshot_download`.
+
+    Returns:
+        The `fastai.Learner` model in the `model_id` repo.
+    """
+    # Check that fastai and fastcore versions are supported.
+    check_fastai_fastcore_versions()
+
+    # Load the `repo_id` repo.
+    # `snapshot_download` returns the folder where the `model_id` repo was stored.
+    # `cache_dir` will be the default '/root/.cache/huggingface/hub'
+    storage_folder = snapshot_download(repo_id=model_id, revision=revision)
+
+    # Check that fastai and fastcore versions in the `model_id` repository are supported.
+    check_fastai_fastcore_pyproject_versions(storage_folder)
+
+    # Loading `model.pkl`.
+    logger.info(
+        f"Using `fastai.Learner` stored in {os.path.join(model_id, 'model.pkl')}."
+    )
+    # Import `load_learner` from `fastai.learner`.
+    from fastai.learner import load_learner
+
+    model = load_learner(os.path.join(storage_folder, "model.pkl"))
+    return model
+
+
+def push_to_hub_fastai(
+    learner,
+    repo_id: str,
+    commit_message: Optional[str] = "Add model",
+    private: Optional[bool] = None,
+    use_auth_token: Optional[Union[bool, str]] = True,
```

Comment:
Thinking about this, there is no scenario in which you can push without a token. Instead, I would change this parameter to be `token: Optional[str] = None`. 

Then, you can do 

```python 
if token is None:
  token = HfFolder.get_token()

if token is None:
    raise ValueError(
        "You must login to the Hugging Face Hub. There are two options:"
        "(1) Type `huggingface-cli login` in your terminal and enter your token."
        "(2) Enter your token in the `token` argument."
        "Your token is available in the Settings of your Hugging Face account."
    )
```

Reply:
Very elegant solution! This simplifies quite a lot!